### DISCUSSION TITLE
Allowing AnimatorState to Specify a RuntimeAnimatorController

### DISCUSSION THREAD

**Discussion by kiygskr** [TylerTemp/SaintsField#113] [Ideas]
Currently, AnimatorState can only specify an Animator, but would it be possible to allow specifying a RuntimeAnimatorController instead?
Looking at the implementation of AnimatorStateAttributeDrawer, it seems to only use Animator.runtimeAnimatorController, so it appears feasible.
If RuntimeAnimatorController could be specified, it would enable AnimatorState to be used with ScriptableObject and other assets, which would be extremely convenient.

**Comment by TylerTemp** [comment]
Hi,

this feature has been added in [3.11.0](https://github.com/TylerTemp/SaintsField/releases/tag/3.11.0)

```csharp
public RuntimeAnimatorController animController;
[AnimatorState(nameof(animController))] public AnimatorState animState;
```

![image](https://github.com/user-attachments/assets/9e0d3067-7e80-4f58-b6e0-7be14ac8ae91)

**Comment by kiygskr** [comment]
Thank you for implementing it! I really appreciate it.
